### PR TITLE
telemetry - adopt `skipTelemetry` in some places (#172640)

### PR DESCRIPTION
**Pull request by bpasero**
@sbatten I am still seeing `workbenchActionExecuted` logged twice even with this change so I would need more help on resolving this.

Here is how I still see telemetry logged:
* you can run out of sources with this change
* set trace level to logging
* open the telemetry output service
* invoke any of the actions below

image

image

I see:

image

### CODE REVIEW

File: src/vs/workbench/services/contextmenu/electron-sandbox/contextmenuService.ts [hunk 1]
```diff
@@ -244,7 +244,9 @@ class NativeContextMenuService extends Disposable implements IContextMenuService
 	}
 
 	private async runAction(actionRunner: IActionRunner, actionToRun: IAction, delegate: IContextMenuDelegate, event: IContextMenuEvent): Promise<void> {
-		this.telemetryService.publicLog2<WorkbenchActionExecutedEvent, WorkbenchActionExecutedClassification>('workbenchActionExecuted', { id: actionToRun.id, from: 'contextMenu' });
+		if (!delegate.skipTelemetry) {
```
Comment (bpasero) [inline src/vs/workbench/services/contextmenu/electron-sandbox/contextmenuService.ts:247]: @sbatten fyi missing adoption?

Comment (sbatten) [reply src/vs/workbench/services/contextmenu/electron-sandbox/contextmenuService.ts:247]: Yea, I didn't realize we did it on native as well for some reason.

**Comment by sbatten** [comment]
For the triple-dot menu, that context menu is shown here. It's much farther down where you aren't really in control of the delegate, so the dropdown probably needs to accept options as well. Maybe there is a cleaner/better stadard way for us to have this 
https://github.com/microsoft/vscode/blob/7ab22649da518ad7a89f0872c1a75ccd130a21f9/src/vs/base/browser/ui/dropdown/dropdown.ts#L206

**Comment by sbatten** [comment]
For the context menu, on the side bar, that's just a context menu so it's really up to the sidebar to decide if it should say context menu or side bar. Maybe this shouldn't be skip telemetry, but instead we should just pass telemetry source all the way down. and it overrides context menu and then the context menu is always logging.

**Comment by sbatten** [comment]
the toolbar is creating an action which is renders as a dropdown menu action which then creates a context menu. the context menu provider is passed through, but the delegate which houses the option to skip telemetry is fully provided by the dropdown menu.

but the right click menu on the side bar is having the same issue, so the toolbar is not the only place where we log telemetry from a source and the context menu logs a second event.

my proposal was to never have the owners (side bar, toolbar, etc) log telemetry for actions that are added to the context menu. instead provide a telemetry source value to override 'contextMenu'

**Comment by bpasero** [comment]
@sbatten 

> my proposal was to never have the owners (side bar, toolbar, etc) log telemetry for actions that are added to the context menu. instead provide a telemetry source value to override 'contextMenu'

Yes, I like that, though I wonder how that would change our telemetry understanding for how actions are being triggered. Because I would assume that eventually we will no longer have `contextMenu` as a source at all, but rather very specific sources wherever the context menu is used.

Irrespective, I still think the toolbar widget needs to change too: I think today it logs telemetry for any action executed but with your proposal we need to stop sending telemetry from the toolbar for actions that are shown in the context menu, no?

**Comment by bpasero** [comment]
Continues in https://github.com/microsoft/vscode/pull/184299

**Pull request closed without merging**